Python/bytecodes.c

```diff
@@ -3955,16 +3953,15 @@ dummy_func(
             assert(self_o != NULL);
             DEOPT_IF(!PyList_Check(self_o));
             STAT_INC(CALL, hit);
-            if (_PyList_AppendTakeRef((PyListObject *)self_o, PyStackRef_AsPyObjectSteal(arg)) < 0) {
-                goto pop_1_error;  // Since arg is DECREF'ed already
-            }
+            int err = _PyList_AppendTakeRef((PyListObject *)self_o, PyStackRef_AsPyObjectSteal(arg));
             PyStackRef_CLOSE(self);
             PyStackRef_CLOSE(callable);
-            STACK_SHRINK(3);
+            ERROR_IF(err, error);
+        #if TIER_ONE
```

Comment:
At first glance, this appears incorrect. The code should either push `None` *or* skip the `POP_TOP`.
Can you add a comment here that the skip is handed for tier 2 during trace projection.